libraries/ESP8266WebServer/src/ESP8266WebServer-impl.h

```diff
@@ -695,49 +695,136 @@ void ESP8266WebServerTemplate<ServerType>::_finalizeResponse() {
-    case 413: return F("Request Entity Too Large");
-    case 414: return F("Request-URI Too Large");
-    case 415: return F("Unsupported Media Type");
-    case 416: return F("Requested range not satisfiable");
-    case 417: return F("Expectation Failed");
-    case 418: return F("I'm a teapot");
-    case 500: return F("Internal Server Error");
-    case 501: return F("Not Implemented");
-    case 502: return F("Bad Gateway");
-    case 503: return F("Service Unavailable");
-    case 504: return F("Gateway Time-out");
-    case 505: return F("HTTP Version not supported");
-    default:  return F("");
-  }
+String ESP8266WebServerTemplate<ServerType>::responseCodeToString(const int code) {
+    const __FlashStringHelper *r;
+    switch (code)
+    {
+    case 100:
+        r = F("Continue");
+        break;
+    case 101:
+        r = F("Switching Protocols");
+        break;
+    case 200:
+        r = F("OK");
+        break;
+    case 201:
+        r = F("Created");
+        break;
+    case 202:
+        r = F("Accepted");
+        break;
+    case 203:
+        r = F("Non-Authoritative Information");
+        break;
+    case 204:
+        r = F("No Content");
+        break;
+    case 205:
+        r = F("Reset Content");
+        break;
+    case 206:
+        r = F("Partial Content");
+        break;
+    case 300:
+        r = F("Multiple Choices");
+        break;
+    case 301:
+        r = F("Moved Permanently");
+        break;
+    case 302:
+        r = F("Found");
+        break;
+    case 303:
+        r = F("See Other");
+        break;
+    case 304:
+        r = F("Not Modified");
+        break;
+    case 305:
+        r = F("Use Proxy");
+        break;
+    case 307:
+        r = F("Temporary Redirect");
+        break;
+    case 400:
+        r = F("Bad Request");
+        break;
+    case 401:
+        r = F("Unauthorized");
+        break;
+    case 402:
+        r = F("Payment Required");
+        break;
+    case 403:
+        r = F("Forbidden");
+        break;
+    case 404:
+        r = F("Not Found");
+        break;
+    case 405:
+        r = F("Method Not Allowed");
+        break;
+    case 406:
+        r = F("Not Acceptable");
+        break;
+    case 407:
+        r = F("Proxy Authentication Required");
+        break;
+    case 408:
+        r = F("Request Timeout");
+        break;
+    case 409:
+        r = F("Conflict");
+        break;
+    case 410:
+        r = F("Gone");
+        break;
+    case 411:
+        r = F("Length Required");
+        break;
+    case 412:
+        r = F("Precondition Failed");
+        break;
+    case 413:
+        r = F("Request Entity Too Large");
+        break;
+    case 414:
+        r = F("URI Too Long");
+        break;
+    case 415:
+        r = F("Unsupported Media Type");
+        break;
+    case 416:
+        r = F("Range not satisfiable");
+        break;
+    case 417:
+        r = F("Expectation Failed");
+        break;
+    case 418:
```

Comment:
Can we drop the silly 418, for another 16 bytes of flash saved?

Reply:
This is RFC2324, and supported by all major browsers.
But yes :)

Reply:
done